src/backends/mysql/blob.cpp

```diff
@@ -18,40 +21,111 @@
 using namespace soci;
 using namespace soci::details;
 
-mysql_blob_backend::mysql_blob_backend(mysql_session_backend &session)
-    : session_(session)
+mysql_blob_backend::mysql_blob_backend(mysql_session_backend &)
+    : details::trivial_blob_backend()
 {
-    throw soci_error("BLOBs are not supported.");
 }
 
 mysql_blob_backend::~mysql_blob_backend()
 {
 }
 
-std::size_t mysql_blob_backend::get_len()
+static unsigned char decode_hex_digit(char c)
 {
-    throw soci_error("BLOBs are not supported.");
+    unsigned char i = static_cast<unsigned char>(tolower(c));
+
+    if (!( (i >= '0' && i <= '9') && (i >= 'a' && i <= 'f') ))
+    {
+        throw soci_error("MySQL BLOB: Encountered invalid characters in hex representation");
+    }
+
+    if (i <= '9')
+    {
+        return i - '0';
+    }
+    else
+    {
+        return i - 'a';
+    }
 }
 
-std::size_t mysql_blob_backend::read_from_start(char * /* buf */, std::size_t /* toRead */, std::size_t /* offset */)
+static char encode_hex_digit(unsigned char d) {
+    static const char hexMap[] = "0123456789abcdef";
+
+    return hexMap[d];
+}
+
+std::size_t mysql_blob_backend::hex_str_size() const
 {
-    throw soci_error("BLOBs are not supported.");
+    // Every byte is represented by 2 hex digits
+    // +2 as for non-empty buffers, the resulting hex sequence is prefixed by "0x"
+    return buffer_.size() * 2 + (buffer_.empty() ? 0 : 2);
 }
 
-std::size_t mysql_blob_backend::write_from_start(char const * /* buf */, std::size_t /* toWrite */, std::size_t /* offset */)
+void mysql_blob_backend::write_hex_str(char *buf, std::size_t size) const
 {
-    throw soci_error("BLOBs are not supported.");
+    assert(size >= hex_str_size());
```

Comment:
Imo exceptions are for (potentially) recoverable errors and violating function contracts is not one of those. In fact in this particular case I also have a code path that throws an exception if the contract is violated, in case a user compiles without assertions (e.g. release mode).
Using the exception has the huge disadvantage of not giving insights where exactly the error occurs (whereas the assertion contains filename and line number)

Reply:
No, sorry, this is not a good reason. It's trivial to stop at exception being thrown while debugging and it's incomparably worse for a library to kill the entire application without any possibility to recover if something goes wrong. Please don't use `assert` in the library code.

Reply:
Again: no downstream user will encounter this in production.
Plus, running things in a debugger is not always possible and sometimes also quite a hassle to break on the _correct_ exception being thrown.

And think about it this way: Unless users explicitly _enable_ assertions (to which I count using a debug build), they won't have them. And if they do enable them, they actually want them.

See also https://stackoverflow.com/a/1245881 form which I like to quote
> Even catching the error and throwing an exception isn't really a solution. The program logic is flawed, and even if we handle the exception, the program is still broken.
> Some errors must be caught during development. If they slip past testing and into the release build used by a customer, the program is just broken, and no amount of runtime error-checking is going to fix it.

Reply:
Nobody explicitly enables the assertions, the vast majority of people just use the IDE which builds in the "debug" mode by default.

Again, killing the application because the author of some library decided that something could never happen is really bad. If it can't happen at all, just remove this. But better throw if only to show that this precondition is supposed to be satisfied.

Reply:
Let's agree to completely disagree on this matter and leave it at that.